Controllers/BranchesController.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using StoresManagement.Data;
+using StoresManagement.Models;
+using StoresManagement.ViewModels;
+
+namespace StoresManagement.Controllers
+{
+    public class BranchesController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IMapper _mapper;
+
+        public BranchesController(ApplicationDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        // GET: Branches
+        public async Task<IActionResult> Index()
+        {
+            var applicationDbContext = _context.Branches.Include(b => b.Contact).Include(b => b.Entity);
+
+            var branches = await applicationDbContext.ToListAsync();
+            var branchesVM = new List<BranchFormViewModel>();
+
+            foreach (var branch in branches)
+            {
+                var branchVM = new BranchFormViewModel();
+
+                branchVM = _mapper.Map<BranchFormViewModel>(branch);
+
+                branchesVM.Add(branchVM);
+            }
+
+            return View(branchesVM);
+        }
+
+        // GET: Entities
+        public async Task<IActionResult> ListBranches(int? id)
+        {
+            var applicationDbContext = _context.Branches
+                .Include(b => b.Entity)
+                .Include(b => b.Contact)
+                .Where(m => m.EntityId == id);
+
+            var branches = await applicationDbContext.ToListAsync();
```

Comment:
Can be simplified
var branches = await _context.Branches
                .Include(b => b.Entity)
                .Include(b => b.Contact)
                .Where(m => m.EntityId == id)
                .ToListAsync();